### ISSUE TITLE
Regular expression syntax

### ISSUE THREAD

**Issue by yrns** [rollup/rollup-plugin-replace#3]
If you use regular expression syntax in the search patterns, you quickly start getting undefined values back as replacements. It's crafty to use a single RegExp and use the patterns as keys to store the replacements, but the returned matches don't necessarily equal the patterns unless they are simple strings. As an example which bit me: searching for `require('path')` requires escaping the parens and quotes, but then the replacement is stored under a key like `'require\\(\'path\'\\)'` when the match returned is simply `require('path')`, which isn't a key in `values`.

I searched for a way to find out what part of the alternation pattern was matched, but I couldn't find anything easy. Otherwise I would have submitted a fix. Maybe using named groups (e.g. https://github.com/edvinv/named-js-regexp) or simply using multiple regular expressions would work. The latter might be slower, but more straightforward.

Or disallow regular expressions and remove any escape sequences in the keys. But that wouldn't be as potentially useful!

**Comment by mbrevda** [comment]
I think the simple and intuitive solution would be to escape all rexeg meta characters before passing `values` to [RegExp](https://github.com/rollup/rollup-plugin-replace/blob/master/src/index.js#L5). Seeing as the escape method already exists, this should be relatively simple. Thoughts?

**Comment by mbrevda** [comment]
for now, I think I'll just go with this [Babel transform](https://babeljs.io/docs/plugins/transform-inline-environment-variables/), which dose the same thing as seems to be way quicker

**Comment by timdp** [comment]
Yeah, I immediately got bitten by the fact that the strings to replace are actually regexps. That's not intuitive.

**Comment by wearhere** [comment]
@Rich-Harris are the values to be replaced _supposed_ to be regexes? I don't see anything about that in the docs; to the contrary, the usage seems to suggest that they should be literal values. So, I think [this line](https://github.com/rollup/rollup-plugin-replace/blob/9d146783b79a7f1e5782661632db1be82f9a0e45/src/index.js#L11) should be
```diff
- const pattern = new RegExp( delimiters[0] + '(' + Object.keys( values ).join( '|' ) + ')' + delimiters[1], 'g' );
+ const pattern = new RegExp( delimiters[0] + '(' + Object.keys( values ).map(escape).join( '|' ) + ')' + delimiters[1], 'g' );
```

**Comment by Rich-Harris** [comment]
This is (finally) implemented in 1.2

**Comment by GerkinDev** [comment]
Not being able to use regular expression is a bit too restrictive IMHO... This could be really useful in some scenarios.
